Headers/DebugServer2/Utils/Bits.h

```diff
@@ -0,0 +1,61 @@
+//
+// Copyright (c) 2014-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the University of Illinois/NCSA Open
+// Source License found in the LICENSE file in the root directory of this
+// source tree. An additional grant of patent rights can be found in the
+// PATENTS file in the same directory.
+//
+
+#ifndef __DebugServer2_Utils_Bits_h
+#define __DebugServer2_Utils_Bits_h
+
+#include <bitset>
+
+#include "DebugServer2/Utils/CompilerSupport.h"
+
+namespace ds2 {
+namespace Utils {
+namespace {
+static inline unsigned int _FFS(unsigned int number) {
+#if __has_builtin(__builtin_ffs)
+  return __builtin_ffs(number);
+#elif defined(_MSC_VER)
+  unsigned long index;
+  if (_BitScanForward(&index, number)) {
+    return index + 1;
+  }
+  return 0;
+#else
+  for (int i = 1; number != 0; i++, number >>= 1) {
+    if (number & 1) {
+      return i;
+    }
+  }
+  return 0;
+#endif
+}
+}
+
+template <typename T> static inline unsigned int FFS(T number) {
+  // We apply the builtin bit functions on chunks of sizeof(int) * 8 bits.
+  const size_t chunkSize = sizeof(int) * 8;
+  while (number != 0) {
+    int ffs = _FFS(number);
+    if (ffs != 0) {
+      return ffs;
+    }
+    number >>= chunkSize;
+  }
+  return 0;
+}
+
+template <typename T> static inline unsigned int PopCount(T number) {
+  std::bitset<sizeof(T) * 8> bits(number);
+  return bits.count();
```

Comment:
As @sas pointed out, your original implementation may be more optimized than this. Code reuse is also a good thing, using `bitset` was a suggestion with that in mind.